packages/account-abstraction/src/SmartAccountAPI.ts

```diff
@@ -185,6 +191,35 @@ export class SmartAccountAPI extends BaseAccountAPI {
    * @param info
    */
   async createUnsignedUserOp(info: TransactionDetailsForBatchUserOp): Promise<UserOperation> {
+    Logger.log('info.paymasterServiceData', info.paymasterServiceData)
+
+    if (
+      info.paymasterServiceData &&
+      info.paymasterServiceData.tokenPaymasterData &&
+      info.paymasterServiceData.tokenPaymasterData.feeTokenAddress &&
+      info.paymasterServiceData.tokenPaymasterData.feeTokenAddress != '' &&
+      info.paymasterServiceData.tokenPaymasterData.feeTokenAddress != ZERO_ADDRESS
+    ) {
+      if (this.paymasterAPI instanceof BiconomyTokenPaymasterAPI) {
+        // update calldata accordingly by checking allowance then batching token approval request
+        const erc20Address = info.paymasterServiceData.tokenPaymasterData.feeTokenAddress
+        Logger.log('erc20 address', erc20Address)
+        const approveTx: Transaction = await this.paymasterAPI.createTokenApprovalRequest(
+          erc20Address,
+          this.provider
+        )
+
+        // May not need to do this always depnding on already given allowance..
```

Comment:
@talhamalik883 @divyaN73 here the code would be plugged in to check existing allowance. then baed on token - transaction payload is requested from token paymaster class to give the allowance data (which can be appended to the batch)